spec/features/staff/users_can_create_a_comment_spec.rb

```diff
@@ -4,74 +4,103 @@
 
-      allow(Activity::VarianceFetcher).to receive(:new).and_return(variance_stub)
-      allow(activity).to receive(:variance_for_report_financial_quarter).with(report: report).and_return(100)
-    end
+        allow(Activity::VarianceFetcher).to receive(:new).and_return(variance_stub)
+        allow(activity).to receive(:variance_for_report_financial_quarter).with(report: report).and_return(100)
+      end
 
-    context "when the user is a BEIS user" do
-      before { authenticate!(user: beis_user) }
+      context "when the user is a BEIS user" do
+        before { authenticate!(user: beis_user) }
 
-      context "when the report is editable" do
-        scenario "the user cannot add a comment" do
-          visit report_path(report)
-          click_on t("tabs.report.variance")
-          expect(page).not_to have_content t("table.body.report.add_comment")
+        context "when the report is editable" do
+          scenario "the user cannot add a comment" do
+            visit report_path(report)
+            click_on t("tabs.report.variance")
+            expect(page).not_to have_content t("table.body.report.add_comment")
+          end
+        end
+
+        context "when the report is not editable" do
+          let(:report) { create(:report, fund: activity.associated_fund, organisation: delivery_partner_user.organisation) }
+          scenario "the user cannot add a comment" do
+            visit report_path(report)
+            click_on t("tabs.report.variance")
+            expect(page).not_to have_content t("table.body.report.add_comment")
+          end
         end
       end
 
-      context "when the report is not editable" do
-        let(:report) { create(:report, fund: activity.associated_fund, organisation: delivery_partner_user.organisation) }
-        scenario "the user cannot add a comment" do
-          visit report_path(report)
-          click_on t("tabs.report.variance")
-          expect(page).not_to have_content t("table.body.report.add_comment")
+      context "when the user is a Delivery Partner user" do
+        before { authenticate!(user: delivery_partner_user) }
+
+        context "when the report is editable" do
+          context "when there are no comments about this activity" do
+            scenario "the user can add a comment" do
+              form = CommentForm.create(report: report)
+
+              expect(form).to have_report_summary_information
+              form.complete(comment: "This activity underspent")
+
+              expect(Comment.all.count).to eq(1)
+              expect(Comment.last.comment).to eq(form.comment)
+
+              expect(page).to have_content t("action.comment.create.success")
+
+              within ".activity_comments" do
+                expect(page).to have_content form.comment
+                expect(page).to have_content I18n.l(Date.today)
+                expect(page).to have_link "#{report.financial_quarter_and_year} #{report.description}"
+              end
```

Comment:
I think your "page object" approach works really well here. It's easy to follow and the fact that you are making assertions in various ways, on:

- the Capybara `form` object
- the Comment model
- the current Capybara `page`

is clear.